tests/bootloader.py

```diff
@@ -0,0 +1,25 @@
+# Copyright (c) 2017
+#
+# All rights reserved.
+#
+# This file is distributed under the Clear BSD license.
+# The full text can be found in LICENSE in the root directory.
+
+import rootfs_boot
+from devices import board, prompt
+
+class UBootVars(rootfs_boot.RootFSBootTest):
+    '''UBootVars prints U-boot environment variables.'''
+    def runTest(self):
+        variables=['loadaddr=0x0E000000', 'stdin=uart@18101500', 'boot_partition=(\d)']
+        board.sendline('fw_printenv | tee uboot_vars && echo \"U-boot environment variables\"')
+        board.expect('U-boot environment variables')
+        board.expect(prompt)
+        for v in variables:
+            board.sendline('egrep -o {} uboot_vars'.format(v))
```

Comment:
Boot_partition variable needs to substitute the \d for [0-9], however then expect(v) won't work since it expects \d... The alternative would be to generate a string containing "grep -q var1 && grep-q var2" based on the vars, and then run "fw_printenv | tee uboot_vars && grep_string && echo ...". Hacky as hell, but should work.